crates/orderbook/src/run.rs

```diff
@@ -411,9 +411,8 @@ pub async fn run(args: Arguments) {
             shutdown_sender.send(()).expect("failed to send shutdown signal");
             match tokio::time::timeout(Duration::from_secs(10), serve_api).await {
                 Ok(inner) => inner.expect("API failed during shutdown"),
-                Err(_) => tracing::error!("API shutdown exceeded timeout"),
+                Err(_) => panic!("API shutdown exceeded timeout"),
```

Comment:
is this panic intended?

Reply:
Yes. We have been shutting down the tokio runtime wrong all this time in the orderbook.
We called `std::process::exit(0)` whether or not the task was shut down in a timely manner. That caused all the tokio tasks to not run their destructor correctly which made it hard to clean up some resources.
Now we only panic when we exceed the timeout and assume a timely shutdown will happen once the main task terminated. This is also how we do it in the driver for example

Reply:
Clear! thanks for the explanation!